addon-manager/tests/src/test/java/org/jboss/forge/addon/manager/ui/ForgeUpdateDistributionCommandTest.java

```diff
@@ -0,0 +1,349 @@
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.test.UITestHarness;
+import org.jboss.forge.arquillian.AddonDependencies;
+import org.jboss.forge.arquillian.archive.AddonArchive;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.forge.furnace.util.Streams;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+@RunWith(Arquillian.class)
+public class ForgeUpdateDistributionCommandTest
+{
+   private static String FORGE_OLD_VERSION = "2.0.0.Final";
+
+   @Deployment
+   @AddonDependencies
+   public static AddonArchive getDeployment()
+   {
+      AddonArchive archive = ShrinkWrap
+               .create(AddonArchive.class)
+               .addBeansXML()
+               .addAsResource(ForgeUpdateDistributionCommandTest.class.getResource(TEST_OLD_FORGE), TEST_OLD_FORGE);
+      return archive;
+   }
+
+   private String forgeHome;
+   private File tempDir;
+
+   private static String TEST_OLD_FORGE = "/forge-old-distribution.zip";
+
+   @Inject
+   private AnnotatedCommandAggregator extension;
+
+   @Inject
+   private Addon addon;
+
+   @Inject
+   private Furnace furnace;
+
+   @Inject
+   private UITestHarness uiTestHarness;
+
+   @Before
+   public void testInitialization() throws Exception
+   {
+      tempDir = OperatingSystemUtils.createTempDir();
+      disablePreviousAddonCommands(furnace);
+      File inputFile = File.createTempFile("forge-old-distribution", ".zip", tempDir);
+      inputFile.deleteOnExit();
+      try (InputStream iStream = getClass().getResourceAsStream(TEST_OLD_FORGE))
+      {
+         try (OutputStream oStream = new FileOutputStream(inputFile))
+         {
+            Streams.write(iStream, oStream);
+         }
+      }
+
+      ZipFile zipFile = new ZipFile(inputFile.getAbsolutePath());
+      String extractedFolderPath = tempDir.getAbsolutePath() + "/extracted-forge";
+      new File(extractedFolderPath).mkdirs();
+      zipFile.extractAll(extractedFolderPath);
+      forgeHome = extractedFolderPath + "/forge-old-distribution";
+      System.setProperty("forge.home", forgeHome);
+      File homeAddonsDir = new File(forgeHome + "/addons");
+      Assert.assertTrue(homeAddonsDir.exists());
+      Assert.assertTrue(homeAddonsDir.listFiles().length == 0);
+      String uiPreviousVersion = getPreviousVersion(addon.getRepository().getRootDirectory().getPath());
+      //TODO: After introducing new DistributionCommand, changeManagerAddonDirectoryToBeOlder needs adjustment (it changed installed.xml and directory)
+      changeManagerAddonDirectoryToBeOlder(addon.getRepository().getRootDirectory(), uiPreviousVersion);
```

Comment:
This method needs to be changed, because now the package.version() is used to determine if the update is needed